Fachportal Padagogik.js

```diff
@@ -88,32 +88,78 @@ function doWeb(doc, url) {
 
 function scrape(doc, url) {
 	var m = url.match(/FId=([\w\d]+)(&|$)/);
-	//e.g. http://www.fachportal-paedagogik.de/fis_bildung/suche/fis_ausg.html?FId=A18196&lart=BibTeX&Speichern=Speichern&senden_an=+E-Mail-Adresse
-	var bibUrl = "/fis_bildung/suche/fis_ausg.html?FId=" + m[1] + "&lart=BibTeX";
-	ZU.doGet(bibUrl, function(text){
+	if (m) {
+		//e.g. http://www.fachportal-paedagogik.de/fis_bildung/suche/fis_ausg.html?FId=A18196&lart=BibTeX
+		var bibUrl = "/fis_bildung/suche/fis_ausg.html?FId=" + m[1] + "&lart=BibTeX";
 		var translator = Zotero.loadTranslator("import");
 		translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");//BibTex translator
-		translator.setString(text);
-		translator.setHandler("itemDone", function(obj, item) {
-			item.attachments.push({
-				title: "Snapshot",
-				document: doc
+		ZU.doGet(bibUrl, function(text){
+			translator.setString(text);
+			translator.setHandler("itemDone", function(obj, item) {
+				finalize(doc, item);
 			});
-			if (item.url) {
+			translator.translate();
+		});
+	} else {
+		var pdfUrl = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
+		Z.debug(pdfUrl);
+
+		/*
+		//fails on http://www.pedocs.de/frontdoor.php?source_opus=9491&la=de
+		Z.debug(ZU.xpathText(doc, '//span[@class="Z3988"]/@title'));
+		var translator = Zotero.loadTranslator("web");
+		translator.setTranslator("05d07af9-105a-4572-99f6-a8e231c0daef");//COinS translator
+		translator.setDocument(doc);//TODO encoding is normally Western and not utf8
```

Comment:
First I tried to use the COinS translator directly, but on http://www.pedocs.de/frontdoor.php?source_opus=9491&la=de this fails 
```
23:56:18 Translation using Fachportal Pädagogik failed: 
         string => TypeError: translate._sandboxManager.sandbox.Array is not a constructor
         stack => Zotero.Translate.Sandbox.Base.loadTranslator/safeTranslator.setHandler/<@chrome://zotero/content/xpcom/translation/translate.js:263:27
         Zotero.Translate.Base.prototype._runHandler@chrome://zotero/content/xpcom/translation/translate.js:1022:20
         Zotero.Translate.Base.prototype._detectTranslatorsCollected@chrome://zotero/content/xpcom/translation/translate.js:1539:3
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1374:30
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:1371:5
         Zotero.Translate.Search.prototype.complete@chrome://zotero/content/xpcom/translation/translate.js:2354:2
         Zotero.Translate.Base.prototype.decrementAsyncProcesses@chrome://zotero/content/xpcom/translation/translate.js:993:4
         Zotero.Translate.Base.prototype._detectTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1523:3
         Zotero.Translate.Base.prototype._detect/<@chrome://zotero/content/xpcom/translation/translate.js:1504:17
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype._detect@chrome://zotero/content/xpcom/translation/translate.js:1503:1
         Zotero.Translate.Base.prototype._getTranslatorsTranslatorsReceived@chrome://zotero/content/xpcom/translation/translate.js:1128:3
         Zotero.Translate.Base.prototype._getTranslatorsGetPotentialTranslators/<@chrome://zotero/content/xpcom/translation/translate.js:1095:28
         Zotero.Translators</this.getAllForType@chrome://zotero/content/xpcom/translation/translator.js:180:4
         Zotero.Translate.Base.prototype._getTranslatorsGetPotentialTranslators@chrome://zotero/content/xpcom/translation/translate.js:1094:1
         Zotero.Translate.Base.prototype.getTranslators@chrome://zotero/content/xpcom/translation/translate.js:1081:4
         Zotero.Translate.Search.prototype.getTranslators@chrome://zotero/content/xpcom/translation/translate.js:2311:9
         Zotero.Translate.Sandbox.Base.loadTranslator/safeTranslator.getTranslators@chrome://zotero/content/xpcom/translation/translate.js:302:12
         Zotero.Translate.Sandbox.Base.loadTranslator/safeTranslator[i]<@chrome://zotero/content/xpcom/translation/translate.js:401:9
         @resource://gre/modules/RemoteAddonsParent.jsm:761:66
         retrieveNextCOinS@C:\Users\zumstein\AppData\Roaming\Mozilla\Firefox\Profiles\eux6gyz4.coding\zotero\translators\COinS.js:105:3
         doWeb@C:\Users\zumstein\AppData\Roaming\Mozilla\Firefox\Profiles\eux6gyz4.coding\zotero\translators\COinS.js:229:3
         Zotero.Translate.Base.prototype._translateTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1222:4
         Zotero.Translate.Web.prototype._translateTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1811:3
         Zotero.Translate.Base.prototype.translate/</<@chrome://zotero/content/xpcom/translation/translate.js:1195:51
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype.translate/<@chrome://zotero/content/xpcom/translation/translate.js:1195:7
         Zotero.Translators</this.get@chrome://zotero/content/xpcom/translation/translator.js:162:4
         Zotero.Translate.Base.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:1192:1
         Zotero.Translate.Web.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:1802:2
         Zotero.Translate.Sandbox.Base.loadTranslator/safeTranslator.translate@chrome://zotero/content/xpcom/translation/translate.js:317:12
         Zotero.Translate.Sandbox.Base.loadTranslator/safeTranslator[i]<@chrome://zotero/content/xpcom/translation/translate.js:401:9
         @resource://gre/modules/RemoteAddonsParent.jsm:761:66
         scrape@Fachportal PÃ¤dagogik:112:3
         doWeb@Fachportal PÃ¤dagogik:73:3
         Zotero.Translate.Base.prototype._translateTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1222:4
         Zotero.Translate.Web.prototype._translateTranslatorLoaded@chrome://zotero/content/xpcom/translation/translate.js:1811:3
         Zotero.Translate.Base.prototype.translate/<@chrome://zotero/content/xpcom/translation/translate.js:1189:58
         Zotero.Translate.Base.prototype._loadTranslator@chrome://zotero/content/xpcom/translation/translate.js:1574:16
         Zotero.Translate.Base.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:1189:4
         Zotero.Translate.Web.prototype.translate@chrome://zotero/content/xpcom/translation/translate.js:1802:2
         _run@chrome://scaffold/content/scaffold.js:423:4
         run@chrome://scaffold/content/scaffold.js:367:4
         oncommand@chrome://scaffold/content/scaffold.xul:1:1
         
         url => http://www.pedocs.de/frontdoor.php?source_opus=9491&la=de
         downloadAssociatedFiles => true
         automaticSnapshots => true
```